framework/include/actions/AddVariableAction.h

```diff
@@ -54,8 +54,10 @@ class AddVariableAction : public MooseObjectAction
   /**
    * determine the variable type given an FEType and number of components
    */
-  static std::string
-  determineType(const FEType & fe_type, unsigned int components, bool is_fv = false);
+  static std::string determineType(const FEType & fe_type,
```

Comment:
Now that I'm looking at it, we can actually remove `components` overall as an input here. It's extraneous and we don't ever actually care about the number of components to determine the time, we just need to know if it's array or not. This may require some changes elsewhere, but that's okay

Go ahead and add a docstring while you're here and make all of this const:

```
/**
 * Determines a variable type
 * @param fe_type The FE type
 * @param is_fv Whether or not the variable is use for finite volume
 * @param is_array Whether or not the variable is an array variable
 */
static std::string determineType(const FEType & fe_type,
                                 const bool is_fv = false,
                                 const bool is_array = false);
```

Reply:
I agree the we can remove n_component here. But I have a concern that this change may require some changes in other moose-based application. I quickly checked Griffin and found several places to be changed.
Perhaps, we can provide two function interface for determineType as:
```
std::string
AddVariableAction::determineType(const FEType & fe_type,
                                 const unsigned int components,
                                 const bool is_fv)                                                                 
{
  bool is_array = (components > 1) ? true : false;
  return determineType(fe_type, is_fv, is_array);

}
std::string
AddVariableAction::determineType(const FEType & fe_type,
                                 const bool is_fv,
                                 const bool is_array)
{
 ...
}
```

Reply:
Go ahead and do:
```
std::string
AddVariableAction::determineType(const FEType & fe_type,
                                 const unsigned int components,
                                 const bool is_fv)                                                                 
{
  return determineType(fe_type, is_fv, components > 1);
}
```

I'll make a separate issue for this so we can deprecate it and remove it in the apps once this change makes it in

Reply:
@loganharbour 
I found an expect error of this approach in compile as:

```
AddLotsOfDiffusion.C:70:17: error: call to 'determineType' is ambiguous
    auto type = AddVariableAction::determineType(fe_type, 1, false);
```
It seems that the second function argument `1` can be regarded as `bool` type. So, the compiler can not tell which function interfaces should be called. 

Do you have any good idea of resolving this error?

Reply:
Go ahead and rename the new method `variableType`.

Add in the old method:
```
mooseDeprecated("AddVariableAction::determineType() is deprecated. Use AddVariableAction::variableType() instead.");
```

And find any occurrences of `determineType()` and use the new mehod